xdsl/transforms/lower_snitch_runtime.py

```diff
@@ -0,0 +1,140 @@
+from abc import ABC
+from xdsl.ir import Operation, SSAValue, MLContext, Attribute
+from xdsl.dialects.builtin import i32, ModuleOp
+from xdsl.dialects import snitch_runtime, func
+from xdsl.passes import ModulePass
+from xdsl.pattern_rewriter import (
+    PatternRewriter,
+    RewritePattern,
+    op_type_rewrite_pattern,
+    PatternRewriteWalker,
+    GreedyRewritePatternApplier,
+)
+from xdsl.dialects.snitch_runtime import tx_id
+
+
+class LowerGetInfoOpToFunc(RewritePattern, ABC):
+    """
+    Rewrite pattern that matches on all SnitchRuntimeGetInfo ops, since they have
+    the same function signature.
+
+    Note: Takes the name of the op and replaces "snrt." with "snrt_" to link with
+    snrt.h
+    """
+
+    @op_type_rewrite_pattern
+    def match_and_rewrite(
+        self, op: snitch_runtime.SnitchRuntimeGetInfo, rewriter: PatternRewriter, /
+    ):
+        rewriter.replace_matched_op(*self.lower(op))
```

Comment:
I would advise against this pattern. It was used in the stencil dialect for testability etc, but is not really useful here. Could you please refactor the `lower` methods into the `match_and_rewrite` implementations?

Reply:
Thanks for the suggestion, should be fixed now!